src/shogun/mathematics/linalg/internal/implementation/Scale.h

```diff
@@ -66,6 +66,20 @@ struct scale
 	static void compute(Matrix A, Matrix B, Matrix C, T alpha, T beta);
 };
 
+/** Specialization of scale for the Native backend */
+template<> template<class Matrix>
+struct scale<Backend::NATIVE, Matrix>
+{
+		typedef typename Matrix::Scalar T;
+		
+		/** Performs the operation B = alpha*A */
+		static void compute(SGVector<T> A, SGVector<T> B, T alpha)
```

Comment:
@lambday both Eigen3 and ViennaCL don't have ```static void compute(T* A, T* B, index_t num_elements, T alpha)``` in the spcializations we have. I was about to add it since it would be easy for moving other ```SGVector::scale``` calls to ```linalg::scale```